File: docs/Final Presentation/src/fib.hs
fib :: Int -> Int
fib n = fib' 0 1 n where
    fib' :: Int -> Int -> Int -> Int
    fib' x0 _ 0 = x0
    fib' x0 x1 n = fib' x1 (x0 + x1) (n - 1)

File: docs/Final Presentation/src/fib.c
int fib(int n){
    return _fib(0, 1, n);
}
static int _fib(int x0, int x1, int n){
    if(n == 0)
        return x0;
    else
        return fib2(x1, x0 + x1, n - 1);
}

